rust/libnewsboat/src/logger.rs

```diff
@@ -203,6 +201,10 @@ impl Logger {
     /// If the message couldn't be written for whatever reason, this function ignores the failure.
```

Comment:
From line 189 onward:
> /// This method is primarily used for logging things received from C++. Since there is no
/// guarantee that the data we got from C++ is valid UTF-8, we can't put it into &str. So we
/// treat it as a binary stream, and log it using this method.

That is a good idea, but we were passing the data via the regular `log()` method, in a `&str` argument which required everything to be valid utf-8.
I think one of the resulting issues can be seen in the stacktrace provided in
https://github.com/newsboat/newsboat/issues/1964#issuecomment-1030247848

I still need to verify that this PR actually fixes the issue (takes some time due to having to setup a reproducer again, and I'm out of time for this weekend)

Reply:
Done, verified by running `go run server.go` with server.go containing:
```
package main

import (
    "fmt"
    "net/http"
    "time"
)

func helloWorld(w http.ResponseWriter, r *http.Request) {
    t := time.Now()
    fmt.Println(t.Format("2006-01-02 15:04:05") + ": " + r.URL.Path)

    if r.URL.Path != "/" {
            http.NotFound(w, r)
            return
    }

    w.Header().Set("Content-Type", "text/xml;charset=ISO-8859-1")

    w.Write([]byte(`<rss version="2.0">`))
    w.Write([]byte(`<channel>`))
    w.Write([]byte(`    <title>` + "iso-8859-1: \xc0\xc1\xc2\xc3\xc5 \xe0\xe1\xe2\xe3\xe4\xe5" + `</title>`))
    w.Write([]byte(`    <link>http://noticias.uol.com.br/</link>`))
    w.Write([]byte(`    <description>timas Notias</description>`))
    w.Write([]byte(`</channel>`))
    w.Write([]byte(`</rss>`))
}

func main() {
    http.HandleFunc("/", helloWorld)
    http.ListenAndServe(":8000", nil)
}
```

and the newsboat urls file containing `http://localhost:8000/`

Without this PR, Newsboat crashes in a logging-related location.
After applying the changes, Newsboat still crashes (when trimming the non-utf-8 title), but at least it gets past the logging-related crash.